stdlib/REPL/src/REPL.jl

```diff
@@ -89,7 +97,7 @@ function softscope!(ex)
     return ex
 end
 
-const repl_ast_transforms = Any[softscope!]
+const repl_ast_transforms = Any[softscope] # defaults for new REPL backends
```

Comment:
It feels like this could be limited to `Function` of `Callable`. Not that it helps much, but if someone wants to pass an object that has its calling overloaded, one can always just wrap it in a closure.

Reply:
I thought about it, but it’s not clear what we would gain by tightening the signature here? Certainly not performance...

Reply:
Fair enough. Doesn't matter I guess.